lib/hero_detail/hero_detail_component.dart

```diff
@@ -24,9 +24,10 @@ class HeroDetailComponent implements OnInit {
 
   @override
   Future<Null> ngOnInit() async {
-    var _id = _routeParams.get('id');
-    var id = int.parse(_id ?? '', onError: (_) => null);
-    if (id != null) hero = await (_heroService.getHero(id));
```

Comment:
Then `hero` stay null and it seems somewhat safe since hero_detail_component.html is not rendered by `*ngIf="hero != null"` in that file.